src/macros.rs

```diff
@@ -824,25 +824,39 @@ macro_rules! clap_app {
 }
 
 macro_rules! impl_settings {
-    ($n:ident, $($v:ident => $c:path),+) => {
-        pub fn set(&mut self, s: $n) {
-            match s {
-                $($n::$v => self.0.insert($c)),+
+    ($settings:ident, $flags:ident,
+        $( $setting:ident($str:expr) => $flag:path ),+
```

Comment:
Can we mark `$str` as literal string instead of `expr`?

Reply:
Also, please rename `str` to `string`.

Reply:
We can, but this is simply internal macro and it doesn't affect anything. `expr` is simply shorter than `literal`

Reply:
Also, why would we use a longer name instead of shorter one? They both quite readable.

Reply:
I think naming it `str` is failing the tests.

Reply:
FYI, macros don't give a damn about keywords, see https://play.rust-lang.org/ . Keywordds are mere idents from their perspective